R/preprocess_data.R

```diff
@@ -6,6 +6,8 @@
 #' @param outcome_colname column name as a string of the outcome variable
 #' @param method methods to preprocess the data, described in `caret::preProcess` (defaut: `c("center","scale")`, use `NULL` for no normalization)
 #' @param remove_nzv whether to remove variables with near-zero variance (default: `TRUE`)
+#' @param remove_zv whether to remove variables with zero variance (default: `TRUE`; only changes the output if `remove_nzv = FALSE`)
```

Comment:
Rather than having separate parameters for `remove_nzv` and `remove_zv`, it may make more sense to specify those with `method` since they are passed to `caret::preProcess(features, method = method)`. I think it'd make the control flow easier to follow.

Reply:
The way I wrote the function this doesn't work because method is only used on continuous features.

Reply:
I see. Parameters should be independent of each other if possible; having dependent parameters is a sign that they should probably be combined. I propose having a single parameter to handle removing nzv, zv, or neither. e.g. `remove_var` could be `'nzv'`, `'zv'` or `NULL`. That would cleanup this [if-else block](https://github.com/SchlossLab/mikropml/pull/197/files#diff-454875bc46900ebf0071274b2abc2a8969c2ad0b868852f03449e1bfb0d6786aR64-R72).

Reply:
done